src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -148,6 +148,13 @@ QgsPostgresConn *QgsPostgresConn::connectDb( QString conninfo, bool readonly, bo
 
   if ( shared )
   {
+
+#ifdef QGISDEBUG
+    // sharing connection between threads is not safe
+    // See http://hub.qgis.org/issues/13141
+    Q_ASSERT ( QApplication::instance()->thread() == QThread::currentThread() );
+#endif
```

Comment:
Note that you don't have to protect `Q_ASSERT ()` by `#ifdef QGISDEBUG` because Q_ASSERT [does nothing if QT_NO_DEBUG was defined during compilation](http://doc.qt.io/qt-4.8/qtglobal.html#Q_ASSERT).